### DISCUSSION TITLE
Delete like vim does

### DISCUSSION THREAD

**Discussion by EricDriussi** [sxyazi/yazi#1544] [1. Q&A] [answered]
Hi there!
Loving the application so far, really great work.

I was wondering if there is any way I can simulate vim's delete behavior.

I would like to, when removing a file, have it available to paste it around after removing it.

In vim, pressing `dd` removes the whole line. Pressing `p` afterwards would paste the deleted (cut) line.

I would expect this keymap to reproduce that behavior:

```
{ on = ["d", "d"], run = ["yank", "remove --force"], desc = "Delete" }
```

Instead, the file is removed but nothing is available to be pasted afterwards.

**Comment by SolitudeSF** [comment]
this is not really practical in the context of file manager. what you want is cut files to be hidden, but that is very minor cosmetic difference with current behaviour.

**Reply by EricDriussi** [reply]
I wouldn't say its cosmetic tbh

If I cut one file, and then cut another, the first one won't be deleted.
While in vim running `dd` on one line and then on another would permanently delete the first one.

I understand most file manager don't behave like this (and that's fair, most people would find that confusing as hell).
I'm not suggesting this should be the default behavior, it's just something that vim users might be used to.

Just for reference, vifm does behave like I'm describing (`dd` deletes a file, running `p` afterwards pastes it back while running `dd` on another file instead will permanently remove the first one, making the second one available for pasting).

**Comment by TobisLee** [comment]
If you mean cut files, you can try this in your `keymap.toml`:
```
[[manager.prepend_keymap]]
on   = ["d", "d"]
run  = "yank --cut"
desc = "Yank selected files (cut)"
```

**Reply by EricDriussi** [reply]
Thanks for your answer!
However, I'm trying to reproduce vim's behavior when cutting/deleting lines.

The `yank --cut` command would leave the file intact if, instead of pasting it, I yank another one.
I'm was wondering if there's any way I can tell Yazi to permanently delete it in that case.

**Reply by SolitudeSF** [reply, marked as answer]
you can write a plugin for that
